packages/calcite-components/src/components/list/list.tsx

```diff
@@ -937,20 +954,44 @@ export class List extends LitElement implements InteractiveComponent, SortableCo
     const toEl = moveTo.element as List["el"];
     const fromElItems = Array.from(fromEl.children).filter(isListItem);
     const oldIndex = fromElItems.indexOf(dragEl);
+    const newIndex = 0;
 
     if (!fromEl) {
       return;
     }
 
+    if (
+      fromEl.canPull?.({
+        toEl,
+        fromEl,
+        dragEl,
+        newIndex,
+        oldIndex,
+      }) === false
+    ) {
+      this.calciteListPullFail.emit({ toEl, fromEl, dragEl, oldIndex, newIndex });
+      return;
+    }
+
+    if (
+      toEl.canPut?.({
+        toEl,
+        fromEl,
+        dragEl,
+        newIndex,
+        oldIndex,
+      }) === false
+    ) {
+      toEl.putFailed({ toEl, fromEl, dragEl, oldIndex, newIndex });
+      return;
+    }
+
     dragEl.sortHandleOpen = false;
 
     this.disconnectObserver();
 
     toEl.prepend(dragEl);
     expandedAncestors(dragEl);
-    const toElItems = Array.from(toEl.children).filter(isListItem);
-    const newIndex = toElItems.indexOf(dragEl);
```

Comment:
Where is `newIndex` updated now? It looks like it's always 0 in the latest changes, where previously, it depended on the placement.

Reply:
It was always 0, we always put at the top of the list when moving lists via keyboard which is why this code wasn't necessary.